source/static.ts

```diff
@@ -14,6 +14,11 @@ export interface InputOptions {
   command?: string;
 }
 
+export interface FunctionOptions {
+  pkgName: string;
+  pkgSeverity?: string;
```

Comment:
There is but I don't think it's very useful. I travel down each json tree to pull out the module_name and severity. npm and yarn have different structures.

yarn audit json
`
      {
        type: 'auditAdvisory',
        data: {
          resolution: {
            id: 1084,
            path: 'mem',
            dev: false,
            optional: false,
            bundled: false
          },
          advisory: {
            findings: [Array],
            id: 1084,
            created: '2019-07-18T21:30:31.935Z',
            updated: '2019-11-19T23:31:37.349Z',
            deleted: null,
            title: 'Denial of Service',
            found_by: [Object],
            reported_by: [Object],
            module_name: 'mem',
            cves: [],
            vulnerable_versions: '<4.0.0',
            patched_versions: '>=4.0.0',
            overview: "Versions of `mem` prior to 4.0.0 are vulnerable to Denial of Service (DoS).  The package fails to remove old values from the cache even after a value passes its `maxAge` property. This may allow attackers to exhaust the system's memory if they are able to abuse the application logging.",
            recommendation: 'Upgrade to version 4.0.0 or later.',
            references: '- [Snyk Report](https://snyk.io/vuln/npm:mem:20180117)',
            access: 'public',
            severity: 'low',
            cwe: 'CWE-400',
            metadata: [Object],
            url: 'https://npmjs.com/advisories/1084'
          }
        }
      }
`
npm audit json

`
      {
        actions: [
          {
            isMajor: true,
            action: 'install',
            resolves: [Array],
            module: 'mem',
            target: '9.0.1'
          }
        ],
        advisories: {
          '1084': {
            findings: [Array],
            id: 1084,
            created: '2019-07-18T21:30:31.935Z',
            updated: '2019-11-19T23:31:37.349Z',
            deleted: null,
            title: 'Denial of Service',
            found_by: [Object],
            reported_by: [Object],
            module_name: 'mem',
            cves: [],
            vulnerable_versions: '<4.0.0',
            patched_versions: '>=4.0.0',
            overview: "Versions of `mem` prior to 4.0.0 are vulnerable to Denial of Service (DoS).  The package fails to remove old values from the cache even after a value passes its `maxAge` property. This may allow attackers to exhaust the system's memory if they are able to abuse the application logging.",
            recommendation: 'Upgrade to version 4.0.0 or later.',
            references: '- [Snyk Report](https://snyk.io/vuln/npm:mem:20180117)',
            access: 'public',
            severity: 'low',
            cwe: 'CWE-400',
            metadata: [Object],
            url: 'https://npmjs.com/advisories/1084'
          }
        },
        muted: [],
        metadata: {
          vulnerabilities: { info: 0, low: 1, moderate: 0, high: 0, critical: 0 },
          dependencies: 3,
          devDependencies: 0,
          optionalDependencies: 0,
          totalDependencies: 3
        },
        runId: 'dc51e255-bb47-4fa3-a167-030803791616'
      }
`

Reply:
Yeah fair point, perhaps we just chuck the full data object in, so at least it would be possible for someone to traverse the data structure themselves if needed.

Reply:
I would say providing the whole thing would be messy for the end user to traverse. Are you suggesting to only provide the 'advisory' section for yarn and the '1084' part for npm?

Reply:
Or how about providing the following options? Where packageData is just whatever * audit json returns for that package?

{
packageName: module_name
packageSeverity: severity
packageData: data
}

Reply:
Yes that is what I was meaning 👍

Reply:
What type would I use for packageData?

Reply:
Would probably be useful to map out the yarn and npm JSON structures then have it be a union of the 2, you could then use that when parsing the JSON as well, to make all the traversal more type safe as well.